sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -50,6 +50,17 @@ object ParserUtils {
     }
   }
 
+  /** Check if duplicate exist in a set of column names. */
+  def checkDuplicateNames[T](colNames: Seq[String], ctx: ParserRuleContext): Unit = {
+    if (colNames.length != colNames.distinct.length) {
+      val duplicateColumns = colNames.groupBy(identity).collect {
+        case (x, ys) if ys.length > 1 => "\"" + x + "\""
+      }
+      throw new ParseException(s"Column repeated in partitioning column(s) or duplicate column " +
+        s"name key in the table definition: ${duplicateColumns.mkString("[", ",", "]")}", ctx)
```

Comment:
@andrewor14 Let me know if you want me to split it into two separate error messages. That means, we will have an extra checking condition with an extra parameter. Thanks!